server/src/main/java/org/opensearch/action/search/SearchResponse.java

```diff
@@ -417,6 +422,29 @@ public static SearchResponse innerFromXContent(XContentParser parser) throws IOE
                         }
                     }
                     clusters = new Clusters(total, successful, skipped);
+                } else if (EXT_FIELD.match(currentFieldName, parser.getDeprecationHandler())) {
+                    extBuilders = new ArrayList<>();
+                    String extSectionName = null;
+                    while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+                        if (token == XContentParser.Token.FIELD_NAME) {
+                            extSectionName = parser.currentName();
+                        } else {
+                            SearchExtBuilder searchExtBuilder = parser.namedObject(SearchExtBuilder.class, extSectionName, null);
+                            if (!searchExtBuilder.getWriteableName().equals(extSectionName)) {
+                                throw new IllegalStateException(
+                                    "The parsed ["
+                                        + searchExtBuilder.getClass().getName()
+                                        + "] object has a "
+                                        + "different writeable name compared to the name of the section that it was parsed from: found ["
+                                        + searchExtBuilder.getWriteableName()
+                                        + "] expected ["
+                                        + extSectionName
+                                        + "]"
+                                );
+                            }
+                            extBuilders.add(searchExtBuilder);
```

Comment:
I just checked, and this code will get called from the `RestHighLevelClient`'s `search()` method.

I'm trying to think of what will happen if a client that doesn't have a given `SearchExtBuilder` registered to its `NamedXContentRegistry` gets a `SearchReponse` with the serialized instance of that `SearchExtBuilder`. Will it throw an exception?

I honestly don't know, but we should try writing a little program to check. 

@austintlee, are you able to do that? If not, I can give it a shot.

Reply:
I think it'll get to this point:

https://github.com/opensearch-project/OpenSearch/blob/9b9158e4cb692dfe04315225ce2b5f9c6c453307/libs/core/src/main/java/org/opensearch/core/xcontent/NamedXContentRegistry.java#L161

I think it's fairly easy to repro using a unit test.  But to do that, I'd have to have a test class that does not have my test ext builder not registered.

I can add a try-catch block to swallow this?

E.g.

```
SearchExtBuilder searchExtBuilder;
try {
    searchExtBuilder = parser.namedObject(SearchExtBuilder.class, extSectionName, null);
} catch (XContentParseException e) {
  parser.skipChildren():
}
```

One thing that is not clear to me is this part:

https://github.com/opensearch-project/OpenSearch/blob/e15177616803c6d00a52d54753761d9a6b7e940b/client/rest-high-level/src/main/java/org/opensearch/client/RestHighLevelClient.java#L2331

How does the REST client know about all the plugins?  Is that referring to the plugins in the OS core only?

Reply:
I did some more looking into the RestHighLevelClient and how it evolved over time.

If one really wants to use the RestHighLevelClient to parse these Exts, they could add a library dependency on the plugin from their application (not ideal, but fine) and do something like:
```
List<NamedXContentRegistry.Entry> registryEntries = plugin.getSearchExts().stream()
    .map(e -> new NamedXContentRegistry.Entry(SearchExtBuilder.class, e.getName(), e.getParser()))
    .collect(Collectors.toList());
RestClientBuilder restClientBuilder = RestClient.builder(new Node(new HttpHost("localhost", 9200)));
RestHighLevelClient pluginAwareClient = new RestHighLevelClient(restClientBuilder, registryEntries) {};
```

In practice, given that we're trying to encourage folks to move away from the RestHighLevelClient (because using server-side business object classes in your client library violates, like, *every* rule of maintainable API design), I'm inclined to not to worry about it to much. 

I think your try-catch suggestion would address any risk of breaking clients.

Reply:
I see.  Yeah, that's essentially how I had to set up my unit test to test my code.

Reply:
@reta ^^

Reply:
Thanks @austintlee , @msfroh 

> I think your try-catch suggestion would address any risk of breaking clients (and probably doesn't warrant any special testing -- we can see from the code that a client without the added registry entries would throw that exception and it would then get caught).

I think this solution hides the problem, silently catching (and even logging which we don't do in model classes) goes unnoticed easily. For 3.0.0, we could align the request and response behavour - thrown an exception there. For 2.x, we need different approach unless we can guarantee that request and response `ext` sections goes side by side: what I mean by that, if request `ext` went through, the response `ext` will go through as well.

Reply:
@reta the `ext` section is not meant to be symmetric as it is tied to search processors and a search processor is either a request processor (which consumes `ext` in SearchRequest) or a response processor (which produces `ext` in SearchResponse).  A client may be making a search request (to a search pipeline) that contains an `ext` section that is intended for a `SearchPipelinePlugin`, but it does not mean it expects a corresponding `ext` in the response (unless the pipeline has a response processor that writes stuff its `ext` section).

What Michael's suggestion above tries to accomplish is that at least for the RestHighLevelClient, you get back a search response even if you don't get the ext section.

Should I take a crack at a `ErrorLoggingExtBuilder` and update this PR?

Btw, I will remove Logger from SearchResponse in my next commit.

Reply:
> What Michael's suggestion above tries to accomplish is that at least for the RestHighLevelClient, you get back a search response even if you don't get the ext section.

I think we have found the solution to that solves the "unsolvable" case with GenericSearchExtBuilder (thanks to @msfroh )